source/Ordinals/MultiplicationProperties.lagda

```diff
@@ -0,0 +1,489 @@
+        l₂ : f (a₀ , b) ≼⟨ α ×ₒ γ ⟩ f (a , b)
+        l₂ = simulations-are-monotone _ _
+              f sim (a₀ , b) (a , b) (×ₒ-≼-left α β l₁)
+      IV (a' , c') (inr (r , q)) =
+       transport (λ - → - ≺⟨ α ×ₒ γ ⟩ f  (a , b)) eq
+                 (order-pres (a' , b) (a , b) (inr (refl , q)))
+        where
+         eq = f  (a' , b)             ＝⟨ IH (a' , b) (inr (refl , q)) ⟩
+              f' (a' , b)             ＝⟨ refl ⟩
+              (a' , pr₂ (f (a₀ , b))) ＝⟨ ap (a' ,_) (r ⁻¹) ⟩
+              (a' , c')               ∎
+
+\end{code}
+
+The following result states that multiplication for ordinals can be
+cancelled on the left. Interestingly, Andrew Swan [Swa18] proved that
+the corresponding result for mere sets is not provable constructively
+already for α = 𝟚: there are toposes where the statement
+
+  𝟚 × X ≃ 𝟚 × Y → X ≃ Y
+
+is not true for certain objects X and Y in the topos.
+
+[Swa18] Andrew Swan
+        On Dividing by Two in Constructive Mathematics
+        2018
+        https://arxiv.org/abs/1804.04490
+
+\begin{code}
+
+×ₒ-left-cancellable : (α β γ : Ordinal 𝓤)
+                    → 𝟘ₒ ⊲ α
+                    → (α ×ₒ β) ＝ (α ×ₒ γ)
+                    → β ＝ γ
+×ₒ-left-cancellable {𝓤} α β γ (a₀ , a₀-least) =
+ transfinite-induction-on-OO P II β γ
+  where
+   P : Ordinal 𝓤 → 𝓤 ⁺ ̇
+   P β = (γ : Ordinal 𝓤) → (α ×ₒ β) ＝ (α ×ₒ γ) → β ＝ γ
+
+   I : (β γ : Ordinal 𝓤)
+     → (α ×ₒ β) ＝ (α ×ₒ γ)
+     → (b : ⟨ β ⟩) → Σ c ꞉ ⟨ γ ⟩ , (α ×ₒ (β ↓ b) ＝ α ×ₒ (γ ↓ c))
+   I β γ e b = c , eq
+    where
+     𝕗 : (α ×ₒ β) ⊴ (α ×ₒ γ)
+     𝕗 = ≃ₒ-to-⊴ (α ×ₒ β) (α ×ₒ γ) (idtoeqₒ _ _ e)
+     f : ⟨ α ×ₒ β ⟩ → ⟨ α ×ₒ γ ⟩
+     f = [ α ×ₒ β , α ×ₒ γ ]⟨ 𝕗 ⟩
+
+     c : ⟨ γ ⟩
+     c = pr₂ (f (a₀ , b))
+
+     eq = α ×ₒ (β ↓ b)                ＝⟨ 𝟘ₒ-right-neutral (α ×ₒ (β ↓ b)) ⁻¹ ⟩
+          (α ×ₒ (β ↓ b)) +ₒ 𝟘ₒ        ＝⟨ ap ((α ×ₒ (β ↓ b)) +ₒ_) a₀-least ⟩
+          (α ×ₒ (β ↓ b)) +ₒ (α ↓ a₀)  ＝⟨ ×ₒ-↓ α β ⁻¹ ⟩
+          (α ×ₒ β) ↓ (a₀ , b)         ＝⟨ eq₁ ⟩
+          (α ×ₒ γ) ↓ (a₀' , c)        ＝⟨ eq₂ ⟩
+          (α ×ₒ γ) ↓ (a₀ , c)         ＝⟨ ×ₒ-↓ α γ ⟩
+          (α ×ₒ (γ ↓ c)) +ₒ (α ↓ a₀)  ＝⟨ ap ((α ×ₒ (γ ↓ c)) +ₒ_) (a₀-least ⁻¹) ⟩
+          (α ×ₒ (γ ↓ c)) +ₒ 𝟘ₒ        ＝⟨ 𝟘ₒ-right-neutral (α ×ₒ (γ ↓ c)) ⟩
+          α ×ₒ (γ ↓ c)                ∎
+      where
+       a₀' : ⟨ α ⟩
+       a₀' = pr₁ (f (a₀ , b))
+
+       eq₁ = simulations-preserve-↓ (α ×ₒ β) (α ×ₒ γ) 𝕗 (a₀ , b)
+       eq₂ = ap ((α ×ₒ γ) ↓_)
+                (simulation-product-decomposition α β γ (a₀ , a₀-least) 𝕗 a₀ b)
+
+   II : (β : Ordinal 𝓤) → ((b : ⟨ β ⟩) → P (β ↓ b)) → P β
+   II β IH γ e = Extensionality (OO 𝓤) β γ (to-≼ III) (to-≼ IV)
+    where
+     III : (b : ⟨ β ⟩) → (β ↓ b) ⊲ γ
+     III b = let (c , eq) = I β γ  e     b in (c , IH b (γ ↓ c) eq)
+     IV  : (c : ⟨ γ ⟩) → (γ ↓ c) ⊲ β
+     IV  c = let (b , eq) = I γ β (e ⁻¹) c in (b , (IH b (γ ↓ c) (eq ⁻¹) ⁻¹))
+
+\end{code}
+
+Finally, multiplication satisfies the expected recursive equations (which
```

Comment:
I am not requesting any change. I am just wondering. You show that both addition and multiplication satisfy the usual classical equations, which is very nice. Although they don't *define* addition and multiplication in this setting, we may wonder whether these equations have addition and multiplication as unique solutions. I wonder if you've thought about this.

Reply:
Yeah, I think that, for now, it would interesting to add this as comment, because it is a natural question that I am sure many other people will ask themselves.

Reply:
We live in the future now: https://github.com/martinescardo/TypeTopology/pull/322 :)